File: Examples2018/UIRelativeLayout/app/src/main/java/course/examples/ui/relativelayout/RelativeLayoutActivity.java
package course.examples.ui.relativelayout;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.Toast;

public class RelativeLayoutActivity extends Activity {

    private EditText mTextEntry;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        mTextEntry = findViewById(R.id.entry);

    }

    public void cancelButtonClicked(@SuppressWarnings("unused") View view) {
        mTextEntry.setText("");
    }

    public void okButtonClicked(@SuppressWarnings("unused") View view) {
        hideKeyboard();
        Toast.makeText(RelativeLayoutActivity.this,
                R.string.thanks_string,
                Toast.LENGTH_LONG).show();
    }

    private void hideKeyboard() {

        View view = this.getCurrentFocus();
        if (view != null) {
            InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
        }
    }
}

File: ExamplesKotlin/UIRelativeLayout/app/src/main/java/course/examples/ui/relativelayout/RelativeLayoutActivity.kt
package course.examples.ui.relativelayout

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import android.widget.Toast

class RelativeLayoutActivity : Activity() {

    private lateinit var mTextEntry: EditText

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        mTextEntry = findViewById(R.id.entry)

    }

    fun cancelButtonClicked(view: View) {
        mTextEntry.setText("")
    }

    fun okButtonClicked(view: View) {
        hideKeyboard()
        if (mTextEntry.text.isNotEmpty()) {
            Toast.makeText(
                this@RelativeLayoutActivity,
                R.string.thanks_string,
                Toast.LENGTH_LONG
            ).show()
        }
    }

    private fun hideKeyboard() {

        val view = this.currentFocus
        if (view != null) {
            val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }
}